### PR TITLE
add function comments for autograd

### PR DESCRIPTION
**Pull request by joddiy**

### CODE REVIEW

File: python/singa/autograd.py [hunk 1]
```diff
@@ -631,16 +712,39 @@ def backward(self, dy):
 
 
 def add_bias(x, b, axis=0):
+    """
+    Add Bias to each row / column of the Tensor, depending on the axis arg.
+    Args:
+        x: matrix.
+        b (CTensor): bias to be added.
```
Comment (nudles) [inline python/singa/autograd.py:719]: pls try to make the docstring consistent.
e.g., some variables have (CTensor) whereas some do not have.

File: python/singa/autograd.py [hunk 2]
```diff
@@ -631,16 +712,39 @@ def backward(self, dy):
...
-
 class Reshape(Operation):
+    """
+    Reshape the input tensor similar to numpy.reshape. 
+    """
 
     def __init__(self, shape):
+        """
+        Args:
+            shape: Specified shape for output. At most one dimension of the new shape can be -1. In this case,
```
Comment (nudles) [inline python/singa/autograd.py:734]: try to keep  the length of a line within 80.

File: python/singa/autograd.py [hunk 3]
```diff
@@ -697,15 +833,32 @@ def backward(self, dy):
...
+    Performs element-wise binary addition.
+    """
 
     def __init__(self):
         super(Add, self).__init__()
 
     def forward(self, a, b):
+        """
+        Args:
+            a (CTensor): First operand..
```
Comment (nudles) [inline python/singa/autograd.py:857]: For simple functions, we can simplify the docstring, e.g., no need to use "First operand.." or "Second operand". You can put "Return a+b, where a and b are CTensor."

File: python/singa/autograd.py [hunk 4]
```diff
@@ -725,12 +885,26 @@ def backward(self, dy):
 
 
 def add(a, b):
+    """
+    Performs element-wise binary addition.
+    Args:
+        a (CTensor): First operand..
```
Comment (nudles) [inline python/singa/autograd.py:891]: a is Tensor or CTensor?

**Comment by nudles** [comment]
docString is to explain the APIs.
We do not need to make it wordy, just make it clear. 
E.g., for `a + b`, we do not need to have "a" is the first operand. Instead, it is more useful to explain if a and b must be of the same shape, is boradcast done automatically.

File: python/singa/autograd.py (hunk 1 above)
Comment (joddiy) [reply python/singa/autograd.py]: fixed, and I have checked the whole file again.

File: python/singa/autograd.py (hunk 2 above)
Comment (joddiy) [reply python/singa/autograd.py]: fixed, and I have checked the whole file again.

File: python/singa/autograd.py (hunk 3 above)
Comment (joddiy) [reply python/singa/autograd.py]: fixed, and for other simple operators, such as or, and, etc, I also change them like this.

File: python/singa/autograd.py (hunk 4 above)
Comment (joddiy) [reply python/singa/autograd.py]: sorry, I made a mistake, it should be Tensor, I've checked all the file again to fix it.

**Comment by joddiy** [comment]
@nudles  I've checked the whole file again and fixed all the problems. I will create another PR later.

**Pull request closed without merging**